packages/loopring_v3/contracts/core/impl/libtransactions/WithdrawTransaction.sol

```diff
@@ -191,21 +206,16 @@ library WithdrawTransaction
         offset += 20;
         withdrawal.accountID = data.toUint32(offset);
         offset += 4;
-        withdrawal.tokenID = data.toUint16(offset) >> 4;
-        withdrawal.feeTokenID = uint16(data.toUint16(offset + 1) & 0xFFF);
-        offset += 3;
+        withdrawal.tokenID = data.toUint16(offset);
+        offset += 2;
         withdrawal.amount = data.toUint96(offset);
         offset += 12;
+        withdrawal.feeTokenID = data.toUint16(offset);
+        offset += 2;
         withdrawal.fee = uint(data.toUint16(offset)).decodeFloat(16);
         offset += 2;
-        withdrawal.to = data.toAddress(offset);
+        withdrawal.dataHash = data.toBytes20(offset);
```

Comment:
move datahash to the end?

Reply:
The style I tried to follow is to always do the nonce at the end. Any reason you think `datahash` is better at the end here?

Reply:
It seems the dataHash doesn't need to be calculated on client-side (and even if it is provided by the user, we may still ignore the value). The operator can calculate this value before putting the tx into a block. That's why I feel it is very special.

Reply:
Added it to the end.